Utility.cs

```diff
@@ -537,5 +539,11 @@ public static IEnumerable<T> Evaluate<T>(this IEnumerable<T> enumerable)
 				hasGraveyard = true;
 			}
 		}
+
+		public static void SaveModConfig(ModConfig config) {
+			MethodInfo configManagerSaveMethod = typeof(ConfigManager).GetMethod("Save", BindingFlags.Static | BindingFlags.NonPublic, new[] { typeof(ModConfig) }) 
```

Comment:
Suggestion: make `configManagerSaveMethod` a global variable and then use `??=` to assign it a value here in order to save overhead from repeated SaveModConfig usage

Reply:
> Suggestion: make `configManagerSaveMethod` a global variable and then use `??=` to assign it a value here in order to save overhead from repeated SaveModConfig usage

fixed by Expression Tree, this is better than using reflection directly

Reply:
Noted.  Although, I suggest delaying the Expression compiling (getting the `MethodInfo`, building the `Expression`) to when `SaveModConfig` is called.  
Doing this will allow any errors to properly be reported as originating from `SaveModConfig` rather than the type initializer for `Utility`.
```cs
if (ConfigManagerSaveMethod is null) {
    ConfigManagerSaveMethod = typeof(ConfigManager).GetMethod("Save", BindingFlags.Static | BindingFlags.NonPublic, new[] { typeof(ModConfig) })
        ?? throw new InvalidOperationException($"Method \"{typeof(ConfigManager).FullName}}::Save\" does not exist.");
    ModConfigParameter = Expression.Parameter(typeof(ModConfig));
    ConfigManagerSave = Expression.Lambda<Action<ModConfig>>(Expression.Call(ConfigManagerSaveMethod, ModConfigParameter), ModConfigParameter)
        .Compile();
}
```

Reply:
If you do that then only `ConfigManagerSave` needs to be a fields. The method and parameter can be locals

Reply:
> If you do that then only `ConfigManagerSave` needs to be a fields. The method and parameter can be locals

Caching the `MethodInfo` is still preferable, even if Expression is used.

Reply:
Ah, wait, it would only need to be loaded once for the Expression.  I see what you mean.

Reply:
> Noted. Although, I suggest delaying the Expression compiling (getting the `MethodInfo`, building the `Expression`) to when `SaveModConfig` is called. Doing this will allow any errors to properly be reported as originating from `SaveModConfig` rather than the type initializer for `Utility`.
> 
> ```cs
> if (ConfigManagerSaveMethod is null) {
>     ConfigManagerSaveMethod = typeof(ConfigManager).GetMethod("Save", BindingFlags.Static | BindingFlags.NonPublic, new[] { typeof(ModConfig) })
>         ?? throw new InvalidOperationException($"Method \"{typeof(ConfigManager).FullName}}::Save\" does not exist.");
>     ModConfigParameter = Expression.Parameter(typeof(ModConfig));
>     ConfigManagerSave = Expression.Lambda<Action<ModConfig>>(Expression.Call(ConfigManagerSaveMethod, ModConfigParameter), ModConfigParameter)
>         .Compile();
> }
> ```

In fact, the initialization of static fields occurs on first access. So the stack display is also correct.
https://dotnetfiddle.net/YuGTrO

Reply:
Oh, indeed. I didn't consider the scenario of multiple static fields. I'll revise it tonight.